components/Init/src/main.c

```diff
@@ -198,7 +198,9 @@ void pre_init(void)
     }
 
     /* add untyped mmios */
-    for (int i = 0; i < simple_get_untyped_count(&camkes_simple); i++) {
+    int cnt = simple_get_untyped_count(&camkes_simple);
+    assert(cnt >= 0); /* This shall never fail. */
```

Comment:
It might be better to use ZF_LOG_IF(cnt < 0, "some error message") to explicitly abort on failure even if asserts are disabled?

Reply:
Changed.